indico/modules/events/__init__.py

```diff
@@ -275,9 +276,9 @@ def _render_location_field(event, **kwargs):
 
 
 @template_hook('event-category-field')
-def _render_category_field(**kwargs):
+def _render_category_field(category_id, **kwargs):
     from indico.modules.events.forms import EventCategoryForm
-    form = EventCategoryForm()
+    form = EventCategoryForm(obj=FormDefaults(category=Category.get_one(int(category_id))) if category_id else None)
```

Comment:
`is_deleted=False` in the get_one call. And I think it's more readable if you move it into a separate line/statement (`category = ...`)